components/board/client.go

```diff
@@ -245,12 +249,52 @@ func (dic *digitalInterruptClient) Tick(ctx context.Context, high bool, nanoseco
 	panic(errUnimplemented)
 }
 
-func (dic *digitalInterruptClient) AddCallback(c chan Tick) {
+func (dic *digitalInterruptClient) AddCallback(ch chan Tick) {
 	panic(errUnimplemented)
 }
 
-func (dic *digitalInterruptClient) RemoveCallback(c chan Tick) {
-	panic(errUnimplemented)
+func (dic *digitalInterruptClient) RemoveCallback(ch chan Tick) {
+}
+
+func (c *client) StreamTicks(ctx context.Context, interrupts []string, ch chan Tick, extra map[string]interface{}) error {
+	ext, err := protoutils.StructToStructPb(extra)
+	if err != nil {
+		return err
+	}
+	stream := &interruptStream{
+		extra:  ext,
+		client: c,
+	}
+
+	c.mu.Lock()
+	c.interruptStreams = append(c.interruptStreams, stream)
+	c.mu.Unlock()
+
+	err = stream.startStream(ctx, interrupts, ch)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *client) removeStream(s *interruptStream) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for i := range s.interruptStreams {
+		if c.interruptStreams[i] == s {
+			// To remove this item, we replace it with the last item in the list, then truncate the
+			// list by 1.
+			s.client.interruptStreams[i] = s.client.interruptStreams[len(s.client.interruptStreams)-1]
+			s.client.interruptStreams = s.client.interruptStreams[:len(s.client.interruptStreams)-1]
+			break
+		}
+	}
+}
+
+// RemoveCallbacks removes the callbacks from the given interrupts.
+func (c *client) RemoveCallbacks(ctx context.Context, interrupts []string, ch chan Tick) error {
+	return nil
```

Comment:
now that this is a helper function rather than an interface on board, please remove it from the board interfaces.